libpod/container_internal_unsupported.go

```diff
@@ -0,0 +1,7 @@
+// +build !linux
+
+package libpod
+
+func (c *Container) cleanupCgroups() error {
+	return ErrOSNotSupported
```

Comment:
`cleanupCgroups` is never going to implemented outside of Linux.  I think it would make more sense to rename it to something like `cleanupResourceLimits` or some such to be the platform-agnostic name for the thing that has to happen [here][1].

[1]: https://github.com/projectatomic/libpod/blob/fd12c8918b0c1e1b481c0ab8573263314fe30db4/libpod/container_internal.go#L930

Reply:
again, my mission is to make it compile with the least amount of change.  stuff like above will get fixed when we work on the actual thing.

Reply:
> I think it would make more sense to rename it to something like `cleanupResourceLimits` or some such...

I've filed the follow-up #1032 to do this.